src/Bridge/Doctrine/Orm/Filter/BooleanFilter.php

```diff
@@ -50,18 +54,31 @@ public function apply(QueryBuilder $queryBuilder, QueryNameGeneratorInterface $q
 
         $properties = $this->extractProperties($request);
 
-        foreach ($properties as $property => $boolean) {
-            if (!$this->isPropertyEnabled($property) || !$this->isPropertyMapped($property, $resourceClass)) {
+        foreach ($properties as $property => $value) {
+            if (
+                !$this->isPropertyEnabled($property) ||
+                !$this->isPropertyMapped($property, $resourceClass) ||
+                !$this->isBooleanField($property, $resourceClass)
+            ) {
                 continue;
             }
 
-            if (empty($boolean) && isset($this->properties[$property])) {
-                $boolean = $this->properties[$property];
-            }
-
-            $filterBoolean = filter_var($boolean, FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE);
+            if (in_array($value, ['true', '1'], true)) {
+                $value = true;
+            } elseif (in_array($value, ['false', '0'], true)) {
+                $value = false;
+            } else {
+                if (null !== $this->logger) {
```

Comment:
I would the constructor like that:

```
public function __construct(ManagerRegistry $managerRegistry, RequestStack $requestStack, LoggerInterface $logger = null, array $properties = null) {
    $this->logger = $logger ?: new NullLogger();
}
```

It eases the maintenance, make the code less complex and is very user friendly even when instantiating the class by hand.
Btw, it helps improving our code coverage :)

Symfony doesn't use this class because when 2.0 was released, PSR-3 wasn't out. The first betas of SF 2 was even using the Zend Log component.